src/actinia_core/core/utils.py

```diff
@@ -49,11 +50,22 @@ def os_path_normpath(path_parts):
         raises a SecurityError if the path is not as expected
     """
     path = os.path.normpath(os.path.join(*path_parts))
-    if not path.startswith(path_parts[0]):
+    if not path.startswith(path_parts[0]) or not path_is_in_allowed_paths(path):
         raise SecurityError(f"Used path '{path}' is insecure")
     return path
 
 
+def path_is_in_allowed_paths(path):
+    # TODO we should double check whether they are all needed, better would be one
+    # single location
+    allowed_starts = [
+        global_config.GRASS_USER_DATABASE,
+        global_config.GRASS_DATABASE,
```

Comment:
Carmen is right that `GRASS_DATABASE` is only okay if read and not if you want to change something in it. Maybe you should still distinguish that somehow?

Reply:
Well it is used in [these lines of PR 280](https://github.com/mundialis/actinia_core/pull/280/files#diff-ee1d2bbb5ac0ad6fc98c8c9b1f98cb33232ee8d70320349b17362de44c629edcR217), that is why I included it.

Reply:
I am not sure if it makes sense to add some `r`/`w` parameter to the functions and split the `allowed_starts`.

Reply:
Ah, I now understand what you mean. I can update this as suggested.

Reply:
I have updated the PR; `ensure_valid_path` now accepts an optional second parameter, the intent ("r", "w" or "rw", Default is "r")